### PR TITLE
starers and fungal infected ferals

### LINKED ISSUE #48438: fungal zombie to scorched zombie
# Describe the bug

when you burn a fungal zombie it becomes a scorched zombie that can revive

# Steps To Reproduce

burn an unpulped fungal zombie

# Expected behavior

fungal zombies when burnt do not become scorched zombies

Do not take a screenshot of the crashing window. Please provide the crash logs in the "Additional context" -->

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1909
- Game Version: 0.E-10281-ge365124 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions],
    Simple_additions [simple_additions],
    Alternative Map Key [alt_map_key],
    C.R.I.T Expansion Mod [crt_expansion],
    Blaze Industries [blazeindustries],
    Stats Through Skills [StatsThroughSkills],
    Bionic Slots [cbm_slots]
]

# Additional context

there is a small  chance this is deliberate but more likely fungal zombies are in the zombie group that can become scorched zombies.

### PR DESCRIPTION
**Pull request by LyleSY**
#### Summary
Content "Adds starers and two fungal infected feral types"

#### Purpose of change

Integrate ferals into the beautiful fungal web of life

#### Describe the solution

A combination of a feral and a burned zombie that keeps some fungal allegiance but can't spread anymore, also a feral with 0 movement, fungal allegiance, no ability to take fungal damage, and a bright future as a fungaloid, plus an intermediate stage that looks like a regular feral but is headed into fungal territory unless it burns into a starer.

#### Describe alternatives you've considered

A similar custom horrifying nonaligned, non-reviving, fungus-proof monster for burned fungal zombies to solve #48438

#### Testing

Loads no errors, look right, behave as expected, explode in smoke and drop weapons
Screen Shot 2021-07-15 at 9 13 01 PM

Screen Shot 2021-07-15 at 9 21 34 PM

New fungal infected ferals are hostile but do nothing as expected. Starers throw rocks and move.
Screen Shot 2021-07-19 at 10 28 35 PM

#### Additional context

"No no no no no no no no no no no no no no." -Radiohead

### CODE REVIEW

**Comment by SariusSkelrets** [comment]
A quick look in the code showed that ferals become normal zombies when they rise again.

Anyway, I have trouble seeing how someone that turned insane before dying then being raised as a walking corpse before being infected by fast-spreading flammable fungus before being killed again by fire would be able to remember their lives to the point pf becoming NPCs

Some charred monster resulting the blob sneaking its way into that burnt yet usable corpse might work better. It also open an option to another monster created when the Mycus takes that corpse back (if such thing is possible)

**Comment by LyleSY** [comment]
Might just be a unique NPC at some point. I like how ridiculously elaborate a background it is.

**Comment by actual-nh** [comment]
> A quick look in the code showed that ferals become normal zombies when they rise again.

It is rather likely that this will be changed.

**Comment by SariusSkelrets** [comment]
Even if it gets changed, I assume them to stay insane or lose some bits of themselves in the lengthy process from feral to starer

And getting filled with flammable matter then roasted enough to stop moving sounds like something that will damage your brain quite a lot

File: data/json/monsters/starers.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
+[
+  {
+    "id": "mon_starer_pipe",
+    "type": "MONSTER",
+    "name": { "str": "starer" },
+    "description": "Burned creature that still reeks of charred flesh.  The flesh has mended into a leathery shell with wide open eyes.  The pupils are dilated and the iris and sclera are bloodshot.  Somehow it still breathes.",
```
Comment (Maleclypse) [inline data/json/monsters/starers.json:6]:
```suggestion
    "description": "Burned creature that still reeks of charred flesh.  The flesh has mended into a leathery shell with wide open eyes.  The pupils are dilated and the iris and sclera are bloodshot.  Somehow it breathes again, in deep ragged gulps.",
```

File: data/json/monsters/starers.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      "CAN_OPEN_DOORS",
+      "PATH_AVOID_DANGER_1",
+      "DROPS_AMMO",
+      "NO_FUNG_DMG"
+    ]
+  },
+  {
+    "id": "mon_starer_crowbar",
+    "type": "MONSTER",
+    "description": "Burned creature that still reeks of charred flesh.  The flesh has mended into a leathery shell with wide open eyes.  The pupils are dilated and the iris and sclera are bloodshot.  Somehow it still breathes.",
```
Comment (Maleclypse) [inline data/json/monsters/starers.json:45]:
```suggestion
    "description": "Burned creature that still reeks of charred flesh.  The flesh has mended into a leathery shell with wide open eyes.  The pupils are dilated and the iris and sclera are bloodshot.  Somehow it breathes again, in deep ragged gulps.",

File: data/json/monsters/starers.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    "copy-from": "mon_starer_pipe",
+    "melee_dice": 2,
+    "melee_dice_sides": 6,
+    "death_drops": "starer_death_drops_crowbar"
+  },
+  {
+    "id": "mon_starer_axe",
+    "type": "MONSTER",
+    "description": "Burned creature that still reeks of charred flesh.  The flesh has mended into a leathery shell with wide open eyes.  The pupils are dilated and the iris and sclera are bloodshot.  Somehow it still breathes.",
```
Comment (Maleclypse) [inline data/json/monsters/starers.json:54]:
```suggestion
    "description": "Burned creature that still reeks of charred flesh.  The flesh has mended into a leathery shell with wide open eyes.  The pupils are dilated and the iris and sclera are bloodshot.  Somehow it breathes again, in deep ragged gulps.",

File: data/json/monsterdrops/feral_humans.json [hunk 4]
```diff
@@ -82,5 +82,23 @@
       { "group": "clothing_watch", "prob": 5 },
       { "group": "wallets", "damage": [ 1, 4 ] }
     ]
+  },
+  {
+    "type": "item_group",
+    "subtype": "collection",
+    "id": "starer_death_drops_axe",
+    "entries": [ { "item": "fire_ax", "prob": 100, "damage": [ 2, 4 ] } ]
```
Comment (Maleclypse) [inline data/json/monsterdrops/feral_humans.json:90]: Are partially burnt items spawnable in itemgroups?

**Review by Maleclypse — COMMENTED** [review]
This is neat.  So the fungal infection restarts some brain activity?   Does it need a mon_faction or species?

**Comment by I-am-Erk** [comment]
These are interesting enough that I think we should try to fit them into the lore somehow. That said, I'm not totally clear on these. Where does the fungus come in? Currently these are just using burns_into, aren't they?

Ferals are basically just the same as the player or any other blob infected human, they just got a heavier hit if whatever takes your risk processing and inhibitions away. They shouldn't rise as anything but zombies, under normal circumstances.

File: data/json/monsterdrops/feral_humans.json (hunk 4 above)
Comment (LyleSY) [reply data/json/monsterdrops/feral_humans.json:90]: Ooh, I like that a lot.

Comment (LyleSY) [reply data/json/monsterdrops/feral_humans.json:90]: But no, not yet as far as I can tell

**Comment by Zireael07** [comment]
What are 'starers' supposed to be? At the very least, it needs a translator's note... because if I, an English major, can't figure it out, it's unlikely anyone will :P

**Comment by LyleSY** [comment]
Google translate says "observers" going from English to Russian and back. It's not a very popular English word, but isn't as archaic as "beholder"

**Comment by actual-nh** [comment]
Could you put `Fixes #48438` in the PR?

**Comment by LyleSY** [comment]
Ah, so that was the initial purpose of this PR and did say that, but actually now with all the changes it doesn't do that anymore

**Comment by actual-nh** [comment]
> Ah, so that was the initial purpose of this PR and did say that, but actually now with all the changes it doesn't do that anymore

Chuckle... understand!

**Comment by LyleSY** [comment]
appveyor error doesn't look related

**Comment by I-am-Erk** [comment]
Apologies for how long it's taken me to get back to this.

I am still not that keen on ferals finalizing like zombies do. Ferals are more like the player, so a fungal feral should be in pain and twitching like when the player doesn't treat their spore infection. They would be creepy, but not very threatening.

**Comment by LyleSY** [comment]
No worries. They fungalize like players do and only become starers if they burn in that middle state

**Comment by I-am-Erk** [comment]
Currently they appear to fungalize into just a fungal version of themselves, still mobile and very happy to have mushrooms sprouting from them.

**Comment by kevingranade** [comment]
Another wrinkle here is that fungals -> spores -> fungal bed is going away, fungal bed is only going to spread around fungal structures.

**Comment by I-am-Erk** [comment]
Does that actually change anything Kevin? It means the fungal spore meat-blobs are less dangerous initially... Later we could maybe add infrastructure to have them evolve into fungal terrain and be another way for the infection to spread, though. For this pr though it still makes an infected feral into a way to spread spores away from the fungal bed.

File: data/json/monsters/feral_humans.json [hunk 5]
```diff
@@ -95,7 +97,71 @@
...
-    "death_drops": "feral_humans_death_drops_axe"
+    "death_drops": "feral_humans_death_drops_axe",
+    "fungalize_into": "mon_feral_human_axe_fungal_infected"
+  },
+  {
+    "id": "mon_feral_human_pipe_fungal_infected",
+    "type": "MONSTER",
+    "description": "Delicate fungal stalks sprout in rows from sores in the arms, and the facial expression is pure joy.  Coos softly.",
+    "copy-from": "mon_feral_human_pipe",
+    "default_faction": "fungus",
```
Comment (I-am-Erk) [inline data/json/monsters/feral_humans.json:108]:
```suggestion
    "default_faction": "fungus",
```
These should still be identical to the original ferals at this point. This phase is the same as a player that has been infected with spores and doesn't know it. From a gameplay perspective, a player shouldn't be able to tell that anything has happened to the feral at all, just like they can't tell when they've been infected with fungal spores.

File: data/json/monsters/feral_humans.json [hunk 6]
```diff
@@ -95,7 +97,71 @@
...
+    "type": "MONSTER",
+    "copy-from": "mon_feral_human_axe",
+    "upgrades": { "age_grow": 4, "into": "mon_feral_human_axe_fungal_corpse" },
+    "burn_into": "mon_starer_axe",
+    "extend": { "flags": [ "NO_FUNG_DMG" ] }
+  },
+  {
+    "id": "mon_feral_human_pipe_fungal_corpse",
+    "type": "MONSTER",
+    "description": "Delicate fungal stalks sprout in rows from sores covering the body, and the facial expression is pure joy.  Coos softly.",
```
Comment (I-am-Erk) [inline data/json/monsters/feral_humans.json:136]:
```suggestion
    "description": "Delicate fungal stalks sprout in rows from sores covering the body, and the facial expression is pure joy.  Coos softly.",
```
This is mostly what I'm looking for, but a fungal corpse probably doesn't have a joyful expression. This is the player after their arms have broken and they've watched the last few moments of their life. They're in intense pain.

File: data/json/monsters/feral_humans.json [hunk 7]
```diff
@@ -95,7 +97,71 @@
...
+    "copy-from": "mon_feral_human_pipe",
+    "default_faction": "fungus",
+    "upgrades": { "half_life": 14, "into": "mon_fungaloid" },
```
Comment (I-am-Erk) [inline data/json/monsters/feral_humans.json:139]: This is anyone's guess, personally I think the corpse should just stay a pile of defunct mushroom food and release spores though.

**Review by I-am-Erk — COMMENTED** [review]
I don't feel like we're quite on the same wavelength here.

File: data/json/monsters/feral_humans.json (hunk 5 above)
Comment (LyleSY) [reply data/json/monsters/feral_humans.json]: Yep, that was an error

File: data/json/monsters/feral_humans.json (hunk 7 above)
Comment (LyleSY) [reply data/json/monsters/feral_humans.json:133]: Or something new and horrifying

**Review by I-am-Erk — APPROVED** [review]
Once conflicts are resolved I think this is good to go

**Pull request merged**